src/PointwiseFunctions/AnalyticSolutions/RelativisticEuler/FishboneMoncriefDisk.hpp

```diff
@@ -381,14 +411,19 @@ class FishboneMoncriefDisk {
     IntermediateVariables(double black_hole_mass, double black_hole_spin,
                           double max_pressure_radius,
                           const gr::Solutions::KerrSchild& background_spacetime,
-                          const tnsr::I<DataType, 3>& x, double t) noexcept;
+                          const tnsr::I<DataType, 3>& x, double t,
+                          size_t in_spatial_velocity_index,
+                          size_t in_lorentz_factor_index) noexcept;
 
     DataType r_squared{};
     DataType sin_theta_squared{};
     double angular_momentum{};
     Scalar<DataType> inv_lapse{};
-    tnsr::I<DataType, 3, Frame::Inertial> shift{};
-    tnsr::ii<DataType, 3, Frame::Inertial> spatial_metric{};
+    tuples::tagged_tuple_from_typelist<
+        typename gr::Solutions::KerrSchild::tags<DataType>>
+        kerr_schild_soln{};
+    size_t spatial_velocity_index;
+    size_t lorentz_factor_index;
```

Comment:
In general we are confused by these indices. Can you add some documentation about what this is optimizing? I'm leaving the review on this line but the documentation can be wherever you think best.